app/controllers/answers_controller.rb

```diff
@@ -1,13 +1,19 @@
 class AnswersController < ApplicationController
   def create
-    question = Question.find(params[:answer][:question])
-    answer = question.answers.new(answer_params)
-    answer.user = current_user
-    if answer.save
-      flash[:success] = "You commented!"
-      redirect_to question_path(question)
+    if current_user
```

Comment:
Glad to have this safety measure in place. Maybe we can pull this logic out to it's own action at some point as it'll be repeated elsewhere.